src/core/geometry/qgsorientedbox3d.cpp

```diff
@@ -93,3 +95,57 @@ QVector<QgsVector3D> QgsOrientedBox3D::corners() const
   }
   return cor;
 }
+
+
+QgsVector3D QgsOrientedBox3D::size() const
+{
+  QgsVector3D axis1( mHalfAxes[0], mHalfAxes[1], mHalfAxes[2] );
+  QgsVector3D axis2( mHalfAxes[3], mHalfAxes[4], mHalfAxes[5] );
+  QgsVector3D axis3( mHalfAxes[6], mHalfAxes[7], mHalfAxes[8] );
+  return QgsVector3D( 2 * axis1.length(), 2 * axis2.length(), 2 * axis3.length() );
+}
+
+
+QgsBox3D QgsOrientedBox3D::reprojectedExtent( const QgsCoordinateTransform &ct ) const
+{
+  // reproject corners to destination CRS
+  QVector<QgsVector3D> c = corners();
+  Q_ASSERT( c.count() == 8 );
+  for ( int i = 0; i < 8; ++i )
+  {
+    c[i] = ct.transform( c[i] );
+  }
+
+  // find AABB for the 8 transformed points
+  QgsVector3D v0 = c[0], v1 = c[0];
+  for ( const QgsVector3D &v : c )
+  {
+    if ( v.x() < v0.x() ) v0.setX( v.x() );
```

Comment:
Replace by std::min/std::max?

```suggestion
    v0.setX( std::min( v0.x(), v.x() ) );
```

Reply:
That would work as well, but is there any advantage over the current implementation?

Reply:
readability. But a bit too late